### ISSUE TITLE
LOLE and Black-out hours should the equal if import capacity and flex is zero

### ISSUE THREAD

**Issue by ChaelKruip** [quintel/etmodel#2425]
In [this scenario](https://beta-pro.energytransitionmodel.com/scenarios/691333) I have set the interconnector to zero and there are no batteries. 

I expect the LOLE and Black-out hours to result in the same number of hours but they are 49 and 284 respectively.

Including @AlexanderWirtz @jorisberkhout @DorinevanderVlies

**Comment by AlexanderWirtz** [comment]
Indeed, @ChaelKruip and I noted this the other day when demonstrating the ETM. Very odd. 

Blackout hours should be a subset of LOLE hours, as some LOLE events can be mitigated by imports. If imports are impossible, i.e. the system is an island, there is no difference between LOLE and blackout hours

**Comment by antw** [comment]
The implementation of these two metrics is different:

* **Loss-of-load expectation** takes [a demand curve from ETEngine](https://github.com/quintel/etsource/blob/313c2a31cd5d2e9913a23d724b726c792d195424/gqueries/modules/security_of_supply/lole_demand.gql), [some dispatchable producers](https://github.com/quintel/etsource/blob/313c2a31cd5d2e9913a23d724b726c792d195424/gqueries/modules/security_of_supply/loss_of_load_expectation.gql#L6-L13) whose production is removed from the demand curve (the original commit describes this as "a replacement for capacity credit") and counts the number of hours in which the resulting demand curve exceeds [a certain capacity](https://github.com/quintel/etsource/blob/313c2a31cd5d2e9913a23d724b726c792d195424/gqueries/modules/security_of_supply/dispatchable_electricity_production_capacity.gql).
* **Blackout hours** simply looks at the number of hours in which demand within the merit order exceeds available supply. Unlike LOLE, the blackout hours calculation does not exclude any dispatchable producers.

As far as I can tell, LOLE is a *probability* based on data pulled out of the ETE graph; blackout hours is an actual measurement of what's happening in Merit. Since LOLE is simply using the "total_demand" curve, it may also not be correctly influenced by the dynamic heat and EV curves.

Possibly the LOLE implementation is now out-of-date; Merit modelled neither import nor flexibility when LOLE was added.

See also:

* quintel/etengine#669: Loss of Load Probability should be calculated from the demand curve
* quintel/etsource#713: [Overview] New implementation of Loss of Load Probability distribution

**Comment by AlexanderWirtz** [comment]
Thanks @antw that clears things up. It seems to me we need to use the same method to calculate both metrics. @jorisberkhout had already mentioned that the two methods might be different. 

The LOLE implementation is probably out of date. Can we calculate LOLE with merit? It would need to know about available import capacity.

**Comment by ChaelKruip** [comment]
> It would need to know about available import capacity.

![trump_wrong](https://user-images.githubusercontent.com/1303760/27947126-da46f954-62f4-11e7-989c-8d8def1c28c8.jpeg)

The LOLE measure is independent of import. It compares *local* demand with *local* dispatchable capacity. In my mind, the black-out calculation **without the aid of import** could function as LOLE.

@jorisberkhout do you agree with this interpretation?

**Comment by AlexanderWirtz** [comment]
This issue is more recent and has a more complete description of how the LOLE calculation is not correct or consistent with the blackout hours calculation: https://github.com/quintel/etmodel/issues/2808
